### ISSUE TITLE
AssertionError from conda inspect linkages

### ISSUE THREAD

**Issue by asmeurer** [conda/conda-build#1794]
```
$conda inspect linkages scikit-umfpack
Traceback (most recent call last):
  File "/Users/aaronmeurer/anaconda/bin/conda-inspect", line 6, in <module>
    sys.exit(conda_build.cli.main_inspect.main())
  File "/Users/aaronmeurer/anaconda/lib/python3.5/site-packages/conda_build/cli/main_inspect.py", line 190, in main
    return execute(sys.argv[1:])
  File "/Users/aaronmeurer/anaconda/lib/python3.5/site-packages/conda_build/cli/main_inspect.py", line 179, in execute
    show_files=args.show_files, groupby=args.groupby))
  File "/Users/aaronmeurer/anaconda/lib/python3.5/site-packages/conda_build/api.py", line 251, in inspect_linkages
    show_files=show_files, groupby=groupby)
  File "/Users/aaronmeurer/anaconda/lib/python3.5/site-packages/conda_build/inspect.py", line 260, in inspect_linkages
    output_string += print_linkages(pkgmap[pkg], show_files=show_files)
  File "/Users/aaronmeurer/anaconda/lib/python3.5/site-packages/conda_build/inspect.py", line 102, in print_linkages
    k = sorted(set(depmap.keys()) - {'system', 'not found'})
  File "/Users/aaronmeurer/anaconda/lib/python3.5/site-packages/conda/models/dist.py", line 226, in __lt__
    assert isinstance(other, self.__class__)
AssertionError
```

**Comment by msarahan** [comment]
Looks like a conda 4.3 compatibility.  Thanks for raising it.

**Comment by msarahan** [comment]
What package did you trigger this with?  ```conda inspect linkages``` works fine for the python package on osx - that's part of our tests.  I want to make sure our tests catch this.

**Comment by asmeurer** [comment]
scikit-umfpack (it's in the first line up there :)

**Comment by msarahan** [comment]
Fixed by #1810 - closing.
